c++/src/kj/common.h

```diff
@@ -1928,6 +1936,30 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
     return true;
   }
 
+  inline bool operator<(const ArrayPtr& other) const {
+    size_t comparisonSize = kj::min(size_, other.size_);
+    if constexpr (isIntegral<RemoveConst<T>>() && isUnsigned<RemoveConst<T>>() && sizeof(T) == 1) {
```

Comment:
I think this won't actually match `char` as written. It'll only match `unsigned char`. But as I said before we should treat `char` as unsigned for comparison purposes. (Only `signed char` is actually signed.)

I think the best answer is probably just:

```suggestion
    if constexpr (isSameType<T, char>() || isSameType<T, unsigned char>()) {
```

Please be sure to add a test of comparisons on `Array<char>` with UTF-8 text containing non-ASCII characters. If `char` is incorrectly being treated as signed then you'll see non-ASCII characters sorting before ASCII, which is wrong, they should sort after.

Reply:
This should be fixed now, I also had to use `RemoveCOnst` and `byte`.